orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/Message.kt

```diff
@@ -24,12 +24,50 @@ import com.netflix.spinnaker.orca.Task
 import com.netflix.spinnaker.orca.pipeline.model.Execution
 import com.netflix.spinnaker.orca.pipeline.model.Stage
 import com.netflix.spinnaker.orca.pipeline.persistence.ExecutionRepository
+import java.util.*
 
 /**
  * Messages used internally by the queueing system.
  */
+
+@JsonTypeInfo(use = MINIMAL_CLASS, include = PROPERTY, property = "@class")
+interface Attribute {
+}
+
+data class MaxAttemptsAttribute(val maxAttempts: Int = -1) : Attribute {
+}
+
+data class AttemptsAttribute(var attempts: Int = 0) : Attribute {
+  fun increment() {
+    this.attempts = attempts + 1
+  }
+}
```

Comment:
Why is it a problem to expand the `Attributes` class? Effectively you _are_ expanding it anyway just in a way that the compiler can't see!